test/dotnet.Tests/CommandTests/Run/RunFileTests.cs

```diff
@@ -1024,4 +1028,280 @@ public void UpToDate_InvalidOptions()
             .Should().Fail()
             .And.HaveStdErrContaining(string.Format(CliCommandStrings.InvalidOptionCombination, RunCommandParser.NoCacheOption.Name, RunCommandParser.NoRestoreOption.Name));
     }
+
+    [Fact]
+    public void Api()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        var programPath = Path.Join(testInstance.Path, "Program.cs");
+        File.WriteAllText(programPath, """
+            #!/program
+            #:sdk Microsoft.NET.Sdk
+            #:sdk Aspire.Hosting.Sdk 9.1.0
+            #:property TargetFramework net11.0
+            #:package System.CommandLine 2.0.0-beta4.22272.1
+            #:property LangVersion preview
+            Console.WriteLine();
+            """);
+
+        new DotnetCommand(Log, "run-api")
+            .WithStandardInput(JsonSerializer.Serialize(new RunApiInput.GetProject { EntryPointFileFullPath = programPath, ArtifactsPath = "/artifacts" }, RunFileApiJsonSerializerContext.Default.RunApiInput))
```

Comment:
It feels like it would be good for the tests to actually write out the expected input and output JSON content. That way if something happens with JsonSerializer or RunApiInput/Output to change the expected schema, the tests will not silently accept the difference and pass, due to using the same serializer as the implementation.

If the JSON is some hell of "\n"s, especially on the output side, it might make sense to break it up somehow with a bunch of string concats, or, write it out with real newlines and .Replace with escaped newlines. Just to keep the transformation from the data we are providing, to the expected json for the test, as minimal and self-contained as possible.